crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use contract_extrinsics::ContractArtifacts;
+use contract_transcode::ink_metadata::MessageParamSpec;
+use scale_info::form::PortableForm;
+use std::path::Path;
+
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract message.
+pub struct Param {
+	/// The label of the parameter.
+	pub label: String,
+	/// The type name of the parameter.
+	pub type_name: String,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract function.
+pub struct ContractFunction {
+	/// The label of the constructor.
+	pub label: String,
+	/// If the message accepts any `value` from the caller.
+	pub payable: bool,
+	/// The parameters of the deployment handler.
+	pub args: Vec<Param>,
+	/// The constructor documentation.
+	pub docs: String,
+	/// If the constructor is the default for off-chain consumers (e.g UIs).
+	pub default: bool,
+	/// If the message is allowed to mutate the contract state. true for constructors.
+	pub mutates: bool,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+/// Specifies the type of contract funtion, either a constructor or a message.
+pub enum FunctionType {
+	Constructor,
+	Message,
+}
+
+/// Extracts a list of smart contract messages parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+pub fn get_messages(path: &Path) -> Result<Vec<ContractFunction>, Error> {
+	let cargo_toml_path = match path.ends_with("Cargo.toml") {
+		true => path.to_path_buf(),
+		false => path.join("Cargo.toml"),
+	};
+	let contract_artifacts =
+		ContractArtifacts::from_manifest_or_file(Some(&cargo_toml_path), None)?;
+	let transcoder = contract_artifacts.contract_transcoder()?;
+	let mut messages: Vec<ContractFunction> = Vec::new();
```

Comment:
Can simplify to `Ok(transcoder.metadata().spec().messages().iter().map(|| ContractFunction{}).collect())`

Reply:
Nicer! Changed in both message and contructors